libraries/AP_GyroFFT/AP_GyroFFT.cpp

```diff
@@ -357,9 +357,8 @@ void AP_GyroFFT::init(uint16_t loop_rate_hz)
 
     // finally we are done
     _initialized = true;
-    update_parameters(true);
     // start running FFTs
-    if (start_update_thread()) {
+    if (update_parameters(true) && start_update_thread()) {
```

Comment:
This will allow the user to arm and (try to?) take off, having previously possibly had some filtering which they were relying on to stop their vehicle going insane (and then screwing that up with a bad parameter set).  It will *fail silently* AFAICS!

I don't think that's at all a good idea....